flow/Makefile

```diff
@@ -98,20 +98,40 @@ default: finish
 #
 # ==============================================================================
 
-# Include design and platform configuration
-include $(DESIGN_CONFIG)
+#-------------------------------------------------------------------------------
+# Default target when invoking without specific target.
+.DEFAULT_GOAL := finish
 
-PUBLIC=nangate45 sky130hd sky130hs asap7
+#-------------------------------------------------------------------------------
+# Proper way to initiate SHELL for make
+SHELL          = /bin/bash
+.SHELLFLAGS    = -o pipefail -c
 
-ifneq ($(wildcard $(PLATFORM_BASE)/$(PLATFORM)),)
-  export PLATFORM_DIR = $(PLATFORM_BASE)/$(PLATFORM)
-else ifneq ($(findstring $(PLATFORM),$(PUBLIC)),)
-  export PLATFORM_DIR = ./platforms/$(PLATFORM)
-else ifneq ($(wildcard ../../$(PLATFORM)),)
-  export PLATFORM_DIR = ../../$(PLATFORM)
-else
-  $(error [ERROR][FLOW] Platform '$(PLATFORM)' not found)
-endif
+#-------------------------------------------------------------------------------
+# Setup varialbles to point to root / head of the OpenROAD directory 
+# - the following settings allowed user to point OpenRoad binaries to different
+#   location
+# - default is current install / clone directory
+export OPENROAD_HOME ?= .
+
+#-------------------------------------------------------------------------------
+# Setup variables to point to other location for the following sub directory
+# - designs - default is under current directory
+# - platforms - default is under current directory
+# - work home - default is current directory
+# - utils, scripts, test - default is under current directory
+export DESIGN_HOME   ?= $(OPENROAD_HOME)/designs
+export PLATFORM_HOME ?= $(OPENROAD_HOME)/platforms
+export WORK_HOME     ?= .
+
+export UTILS_DIR     ?= $(OPENROAD_HOME)/util
+export SCRIPTS_DIR   ?= $(OPENROAD_HOME)/scripts
+export TEST_DIR      ?= $(OPENROAD_HOME)/test
+
+#-------------------------------------------------------------------------------
+# Include design and platform configuration
+include $(DESIGN_CONFIG)
+export PLATFORM_DIR  = $(PLATFORM_HOME)/$(PLATFORM)
```

Comment:
if you make this ?= then the design could overwrite that path.

Reply:
All suggestions are good ideas. Will redo...